Utilities/WSIWindow/window_xcb.h

```diff
@@ -0,0 +1,388 @@
+/*
+*--------------------------------------------------------------------------
+* Copyright (c) 2015-2016 Valve Corporation
+* Copyright (c) 2015-2016 LunarG, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Author: Rene Lindsay <rene@lunarg.com>
+*
+*--------------------------------------------------------------------------
+*/
+
+//============================XCB===============================
+#ifdef VK_USE_PLATFORM_XCB_KHR
+
+#ifndef WINDOW_XCB
+#define WINDOW_XCB
+
+//-------------------------------------------------
+#include "WindowImpl.h"
+//#include <xcb/xcb.h>               // XCB only
+//#include <X11/Xlib.h>              // XLib only
+#include <X11/Xlib-xcb.h>            // Xlib + XCB
+#include <xkbcommon/xkbcommon.h>     // Keyboard
+//-------------------------------------------------
+#ifdef ENABLE_MULTITOUCH
+#include <X11/extensions/XInput2.h>  // MultiTouch
+typedef uint16_t xcb_input_device_id_t;
+typedef uint32_t xcb_input_fp1616_t;
+// clang-format off
```

Comment:
Run this through CF.  After you do it once, you can come back, line up things that you really care about and add the CF-off back in, but this is too different from everything else.